tests/test_basic_ops_list.py

```diff
@@ -598,3 +598,23 @@ def test_getattr() -> None:
     assert getattr(cfg, "0") == src[0]
     assert getattr(cfg, "1") == src[1]
     assert getattr(cfg, "2") == src[2]
+
+
+def test_set_valid_union_value() -> None:
+    cfg = OmegaConf.structured(ListUnion)
+    cfg.list = [4, 3.14]
+    assert isinstance(cfg.list._get_node(0), UnionNode)  # type: ignore
```

Comment:
Could be removed since duplicated two lines below?

Reply:
Where is this duplicated? This method is in test_basic_ops_dict and test_basic_ops_list. Maybe test_structured_config?

Reply:
Two lines below there is exactly the same assert, and in-between only a read access to the config so there's no reason for it to change type (unless you actually want to check that executing `cfg.list == [4, 3.14]` doesn't change the node type? If yes why not, it just looked weird to me)

Reply:
Oh shoot true.